File: electro-server/src/main/java/com/electro/entity/product/Brand.java
package com.electro.entity.product;

import com.electro.entity.BaseEntity;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Accessors(chain = true)
@Entity
@Table(name = "brand")
public class Brand extends BaseEntity {
    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "code", nullable = false, length = 35)
    private String code;

    @Column(name = "description", nullable = false)
    private String description;

    @Column(name = "status", nullable = false, columnDefinition = "TINYINT")
    private String status;
}

File: electro-client/src/models/Brand.ts
import BaseResponse from 'models/BaseResponse';

export interface BrandResponse extends BaseResponse {
  name: string;
  code: string;
  description: string | null;
  status: number;
}

export interface BrandRequest {
  name: string;
  code: string;
  description: string | null;
  status: number;
}
